client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+                    else:
+                        # If no pattern is set, all files match
+                        matches_pattern = True
+
+                    # Validate content type matches item type mapping
+                    matches_content_type = (
+                        pres_repr_content_type in CONTENT_TYPE_MAPPING[item_type]  # noqa
+                    )
+
+                    if (
+                        matches_ext
+                        and matches_pattern
+                        and matches_content_type
+                    ):
+                        matching_files.append(file)
+
+                if matching_files:
+                    abs_dir_path = os.path.join(
+                        media_folder_path, item["clip_dir_subpath"]
+                    ).replace("\\", "/")
+
+                    if item["clip_dir_subpath"] == "/":
+                        abs_dir_path = media_folder_path
+
+                    # get extension from first file
+                    repre_ext = os.path.splitext(
+                        matching_files[0])[1].lstrip(".")
+
+                    repre_data = {
+                        "ext": repre_ext,
+                        "name": preset_repre_name,
+                        "files": matching_files,
+                        "content_type": pres_repr_content_type,
+                        # for reviewable checking in next step
+                        "repre_preset_name": preset_repre_name,
+                        "dir_path": abs_dir_path,
+                        "tags": pres_repr_tags,
+                        "custom_tags": pres_repr_custom_tags,
+                    }
+                    # Add optional output name suffix
+                    suffix = item["suffix"]
+                    if suffix and "thumb" not in suffix:
+                        repre_data["outputName"] = suffix
+                        repre_data["name"] += f"_{suffix}"
+
+                    grouped_representations[product_name][
+                        "representations"].append(repre_data)
+
+        # Second pass: create instances for each group
+        for product_name, group_data in grouped_representations.items():
+            if not group_data["representations"]:
+                continue
+
+            representations = group_data["representations"]
+            # skip case where only thumbnail is present
+            if (
+                len(representations) == 1
+                and representations[0]["content_type"] == "thumbnail"
+            ):
+                continue
+
+            # get version from files with use of pattern
+            # and versioning type
+            version = None
+            if pres_versioning == "from_file":
+                version = self._extract_version_from_files(representations)
+            elif pres_versioning == "locked":
+                version = product_preset["locked"]
+
+            # check if product is reviewable
+            reviewable = any(
```

Comment:
This is hard to read and understand what the logic actually is. It has 4 lines but it takes a minute to "maybe" understand, instead of having 20 lines that can be understood in 5 seconds. Just saying.